combatEngine

```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
 };
 
 function getInitiative(){//Helper Function to create an initiative order.
+    var initiativeOrder = [];
     var needInitiative = fighters
```

Comment:
So this is assigning to the variable needInitiative a reference to the same array that fighters is pointing to.

If you want a new array, use `needInitiative = [...fighters]`